express/component_spec.py

```diff
@@ -0,0 +1,328 @@
+                description="The path to the output manifest",
+            )
+        ]
+        return outputs
+
+    @staticmethod
+    def _add_defaults_values(
+        values: t.Union[None, T, t.List[T]],
+        default_values: t.List[T],
+        value_type: t.Type[T],
+    ) -> t.List[T]:
+        """Add default values to a input/output list attribute"""
+
+        if values is None:
+            values = []
+
+        if isinstance(values, value_type):
+            values = [values]
+
+        values.extend(default_values)
+        return values
+
+    def _add_input_args(self):
+        """Add specified component arguments to the input"""
+        for arg_name, arg_info in self.args.items():
+            self.inputs.append(
+                Input(
+                    name=arg_name.strip(),
+                    description=arg_info["description"].strip(),
+                    type=get_kubeflow_type(arg_info["type"].strip()),
+                )
+            )
+
+    def _set_component_run_cmd(self):
+        """
+        Function that return the run command of the kubeflow component
+        Returns:
+            The kubeflow component run command
+        """
+
+        def _add_run_arguments(args: t.List[T]):
+            for arg in args:
+                arg_name = arg.name.replace("-", "_").strip()
+                arg_name_cmd = f'--{arg.name.replace("_", "-")}'.strip()
+
+                if arg_name == "input_manifest":
+                    arg_kfp_value_type = "inputPath"
+                elif arg_name == "output_manifest":
+                    arg_kfp_value_type = "outputPath"
+                else:
+                    arg_kfp_value_type = "inputValue"
+
+                self.command.append(arg_name_cmd)
+                self.command.append({arg_kfp_value_type: arg_name})
+
+        self.command = ["python3", "main.py"]
+        _add_run_arguments(self.inputs)
+        _add_run_arguments(self.outputs)
+
+    def get_specification(self) -> dict:
+        """
+        Function that returns the specifications of the kubeflow component
+        Returns:
+            the kubeflow specification as a dictionary
+        """
+        if not all(
+            [
+                self.name,
+                self.description,
+                self.inputs,
+                self.outputs,
+                self.image,
+                self.command,
+            ]
+        ):
+            raise ValueError("Missing required attributes to construct specification")
+
+        specification = {
+            "name": self.name,
+            "description": self.description,
+            "inputs": [asdict(input_obj) for input_obj in self.inputs],
+            "outputs": [asdict(output_obj) for output_obj in self.outputs],
+            "implementation": {
+                "container": {"image": self.image, "command": self.command}
+            },
+        }
+
+        return specification
+
+
+class ComponentSubset:
```

Comment:
Is there a need for a different class given that we already have a `Subset` class?

Reply:
the other class has a `location` field which is not needed when you define the component

Reply:
Ok, but the location is just defined as a `property`, right?

Reply:
also takes a `base_path` as a input for initialization so I think they're better separated